### ISSUE TITLE
Consider adding `excludeAnnotatedClasses` to UseDataClass.

### ISSUE THREAD

**Issue by serj-lotutovici** [arturbosch/detekt#694]
I found myself in need of an exclude because some classes that are more configuration that data, give these false positives. For example:

```
@Module
class MyModule(
    @get:Provides val someVal: SomeInterface, 
    @get:Provides val otherVal:SomeOtherInterface
)
```

I could just add a suppress here, but it seems wrong because an object of this class will never require an `equals` or `hashCode`.

**Comment by arturbosch** [comment]
I hate annotation-driven-development :(

**Comment by Mauin** [comment]
This addition does make sense for the rule, though. Especially in Android we'll see quite a few of these annotated classes which don't need to be `data class`es.
